src/Serilog/Events/LogEvent.cs

```diff
@@ -192,3 +194,18 @@ internal LogEvent Copy()
             properties);
     }
 }
+
+internal static class LogEventExtensions
+{
+    public static void AddPropertyIfAbsent(this LogEvent evt, ILogEventPropertyFactory factory, string name, object? value, bool destructureObjects = false)
```

Comment:
Since we already end up adding an `internal` property to `LogEvent`, we could just skip that and make this overload of `AddPropertyIfAbsent` an internal method on `LogEvent` itself, for now?

I can see how dependent libraries would benefit from making this accessible, but in that case we'd have to expose _something_ on `LogEvent` anyway, so we might as well keep everything tight/encapsulated for now.

Reply:
done